pkg/generators/clientgen/gen.go

```diff
@@ -376,7 +397,117 @@ func (g *Generator) generateSubInterfaces(ctx *genall.GenerationContext) error {
+					// +genclient:method=ApplyScale,verb=apply,subresource=scale,input=k8s.io/api/autoscaling/v1.Scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:method=CreateScale,verb=create,subresource=scale,input=k8s.io/api/autoscaling/v1.Scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:method=CreateToken,verb=create,subresource=token,input=k8s.io/api/authentication/v1.TokenRequest,result=k8s.io/api/authentication/v1.TokenRequest
+					// +genclient:method=GetScale,verb=get,subresource=scale,result=Scale
+					// +genclient:method=GetScale,verb=get,subresource=scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:method=UpdateApproval,verb=update,subresource=approval,input=k8s.io/api/certificates/v1.CertificateSigningRequest,result=k8s.io/api/certificates/v1.CertificateSigningRequest
+					// +genclient:method=UpdateEphemeralContainers,verb=update,subresource=ephemeralcontainers
+					// +genclient:method=UpdateScale,verb=update,subresource=scale,input=Scale,result=Scale
+					// +genclient:method=UpdateScale,verb=update,subresource=scale,input=k8s.io/api/autoscaling/v1.Scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:noStatus
+					// +genclient:nonNamespaced
+					// +genclient:noVerbs
+					// +genclient:onlyVerbs=create
+					// +genclient:onlyVerbs=create,delete
+					// +genclient:readonly
+					// +genclient:skipVerbs=deleteCollection
+					// +genclient:skipVerbs=get,update
+				*/
+
+				var additionalMethods []internal.AdditionalMethod
+				var skipVerbs []string
+				var onlyVerbs []string
+
+				genclientMarkers := info.Markers[genclientMarker.Name]
+
+				namespaceScoped := info.Markers.Get(nonNamespacedMarker.Name) == nil
+				noVerbs := info.Markers.Get(noVerbsMarker.Name) != nil
+				hasStatus := hasStatusSubresource(info)
+				readOnly := info.Markers.Get(readOnlyMarker.Name) != nil
+				sVerbs := info.Markers.Get(skipVerbsMarker.Name)
+				if sVerbs != nil {
+					val := string(sVerbs.(markers.RawArguments))
+					skipVerbs = strings.Split(val, ",")
+				}
+				oVerbs := info.Markers.Get(onlyVerbsMarker.Name)
+				if oVerbs != nil {
+					val := string(oVerbs.(markers.RawArguments))
+					onlyVerbs = append(onlyVerbs, strings.Split(val, ",")...)
+				}
+
+				if readOnly {
+					onlyVerbs = append(onlyVerbs, genutil.ReadonlyVerbs...)
+				}
+
+				for _, m := range genclientMarkers {
+					gm := m.(genclient)
+
+					if gm.Method != nil {
+						additionalMethod := internal.AdditionalMethod{
+							Method:      gm.Method,
+							Verb:        gm.Verb,
+							Subresource: gm.Subresource,
+							Input:       gm.Input,
+							Result:      gm.Result,
+						}
+
+						if err := verifyAdditionalMethod(additionalMethod); err != nil {
+							root.AddError(err)
+						}
+						additionalMethods = append(additionalMethods, additionalMethod)
+					}
+
+					// parse onlyVerbs marker
+					if len(onlyVerbs) > 0 {
+
+						// Combine the verbs in only Verbs and skip Verbs list.
```

Comment:
nit: this would be a lot more concise and perhaps more readable if we used `sets.String` utilities here